### PR TITLE
Adding routemap functionality

### PR DESCRIPTION
**Pull request by websitescenes**

### CODE REVIEW

**Comment by jerearista** [comment]
test this please

File: spec/unit/puppet/provider/eos_routemap/default_spec.rb [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+      # Struggling with how to test prefetch
+      # it 'sets the provider instance of the managed resource test' do
+      #   subject
+      #   expect(resources['test:10'].provider.name).to eq('test:10')
+      #   expect(resources['test:10'].provider.description).to eq(@description)
+      #   expect(resources['test:10'].provider.action).to eq(@action)
+      #   expect(resources['test:10'].provider.match).to eq(@match)
+      #   expect(resources['test:10'].provider.set).to eq(@set)
+      #   expect(resources['test:10'].provider.continue).to eq(@continue)
+      # end
```
Comment (websitescenes) [inline spec/unit/puppet/provider/eos_routemap/default_spec.rb]: Still working on this test; may have a bigger issue here.

**Comment by websitescenes** [comment]
retest this please

**Comment by websitescenes** [comment]
retest this please

**Comment by jerearista** [comment]
retest this please

**Pull request merged**